#!/usr/bin/env python
PACKAGE = "cell_culture"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()


target_enum = gen.enum([ gen.const("None",      int_t, 0, "No target"),
                        gen.const("plate_tip_well",     int_t, 1, "Pipette tip to desired well"),
                        gen.const("plate_cam_well",      int_t, 2, "Camera center to desired well"),
                        gen.const("plate_cam_center",      int_t, 3, "Camera center to center of plate"),
                        gen.const("april_tag",      int_t, 4, "Camera center to April Tag"),],
                        "Set perception target")

plate_type_enum = gen.enum([ gen.const(96,      int_t, 0, "96 well plate"),
                        gen.const(24,     int_t, 1, "24 well plate"),
                        gen.const(6,      int_t, 2, "6 well plate"),],
                        "Set plate type")

gen.add("target", int_t, 0, "Set perception target", 0, 0, 4, edit_method=target_enum)
gen.add("plate_type", int_t, 0, "Set plate type", 0, 0, 2, edit_method=plate_type_enum)
gen.add("desired_well",    int_t,    0, "Desired well to servo towards", 0,  0, 95)
gen.add("april_tag_id",    int_t,    0, "April Tag ID to perceive", 0,  0, 20)
gen.add("plate_length_offset", double_t, 0, "Offset the known length dimension of the plate",    0, -10,   10)
gen.add("plate_width_offset", double_t, 0, "Offset the known width dimension of the plate",    0, -10,   10)
gen.add("plate_left_correction", double_t, 0, "Offset the starting left value",    0, -10,   10)
gen.add("plate_up_correction", double_t, 0, "Offset the starting up value",    0, -10,   10)
gen.add("x_tag_offset", double_t, 0, "Offset the x coordinate of the detected April Tag (px)", 0, -400, 400)
gen.add("y_tag_offset", double_t, 0, "Offset the y coordinate of the detected April Tag (px)", 0, -400, 400)
gen.add("z_tag_offset", double_t, 0, "Offset height of the end effector when detecting April Tag (m)", 0, -0.4, 0.4)
gen.add("using_homography",   bool_t,   0, "Using homography to correct camera perspective shift",  False)

 
exit(gen.generate(PACKAGE, "cell_culture", "CellCulturePerception"))